src/main/java/frc/robot/poseestimator/GBPoseEstimator.java

```diff
@@ -0,0 +1,122 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.interpolation.TimeInterpolatableBuffer;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveWheelPositions;
+import frc.robot.poseestimator.observations.OdometryObservation;
+import frc.robot.poseestimator.observations.VisionObservation;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class GBPoseEstimator implements IPoseEstimator {
+
+	private final TimeInterpolatableBuffer<Pose2d> odometryPoseInterpolator;
+	private final SwerveDriveKinematics kinematics;
+	private final double[] odometryStandardDeviations;
+	private Pose2d odometryPose;
+	private Pose2d estimatedPose;
+	private SwerveDriveWheelPositions lastWheelPositions;
+	private Rotation2d lastGyroAngle;
+	private VisionObservation lastVisionObservation;
+
+	public GBPoseEstimator(
+		SwerveDriveKinematics kinematics,
+		SwerveDriveWheelPositions initialWheelPositions,
+		Rotation2d initialGyroAngle,
+		double[] odometryStandardDeviations
+	) {
+		this.odometryPose = new Pose2d();
+		this.estimatedPose = new Pose2d();
+		this.odometryPoseInterpolator = TimeInterpolatableBuffer.createBuffer(PoseEstimatorConstants.POSE_BUFFER_SIZE_SECONDS);
+		this.kinematics = kinematics;
+		this.lastWheelPositions = initialWheelPositions;
+		this.lastGyroAngle = initialGyroAngle;
+		this.odometryStandardDeviations = new double[3];
```

Comment:
Maybe add a record for deviations

Reply:
no need for that after I made an enum for them